tests/backstop_data/gulp-tasks/backstopjs.js

```diff
@@ -396,32 +403,85 @@ function runBackstopJS (command) {
  * @returns {Promise} An empty promise that is resolved when the task is done.
  */
 function setupData () {
-  var activeCaseId = getActiveCaseId();
   var caseType = createUniqueCaseType({
     name: RECORD_IDENTIFIERS.emptyCaseTypeName,
     case_type_category: 'Cases',
-    title: 'Backstop Empty Case Type',
+    title: 'Backstop Case Type',
     definition: {
-      activityTypes: [],
+      activityTypes: [{
+        name: 'Open Case',
+        max_instances: '1'
+      }, {
+        name: 'Follow up'
+      }, {
+        name: 'File Upload'
+      }],
       activitySets: [],
-      caseRoles: [{ name: 'Case Coordinator is', manager: '1', creator: '1' }],
+      caseRoles: [
+        {
+          name: 'Homeless Services Coordinator',
+          creator: '1',
+          manager: '0'
+        }, {
+          name: 'Health Services Coordinator',
+          manager: '0'
+        }, {
+          name: 'Benefits Specialist',
+          manager: '1'
+        }
+      ],
       timelineActivityTypes: []
     }
   });
+
   var activeContact = createUniqueContact({
     contact_type: 'Individual',
     display_name: RECORD_IDENTIFIERS.activeContactDisplayName
   });
+  createUniqueEmail({
+    contact_id: activeContact.id,
+    email: RECORD_IDENTIFIERS.activeContactEmail
+  });
   var emptyContact = createUniqueContact({
     contact_type: 'Individual',
-    display_name: RECORD_IDENTIFIERS.emptyContactDisplayName
+    display_name: RECORD_IDENTIFIERS.emptyContactDisplayName,
+    email: RECORD_IDENTIFIERS.emptyContactEmail
+  });
+  createUniqueEmail({
+    contact_id: emptyContact.id,
+    email: RECORD_IDENTIFIERS.emptyContactEmail
+  });
+
+  var caseIds = createCases(caseType, activeContact, emptyContact);
+  createActivities(caseIds[0], activeContact);
```

Comment:
I have some mixed feelings about these functions. On one hand they look cleaner than everything else, but on the other hand they don't seem to be doing the same thing as the other functions, the function parameters are arbitrary, they create an arbitrary number of cases and activities, and `createCases` also creates an empty case record on top of the 17 others it creates.

I think this approach would be a bit more verbose, but in line with what's happening in this function:

```js
var SAMPLE_NUMBER_OF_CASES = 17;
var SAMPLE_NUMBER_OF_ACTIVITIES = 30;
// ...
var cases = _.range(SAMPLE_NUMBER_OF_CASES).map((index) => createUniqueCase({
  case_type_id: caseType.id,
  contact_id: activeContact.id,
  creator_id: activeContact.id,
  subject: RECORD_IDENTIFIERS.caseSubject + index
});

createUniqueCase({
  case_type_id: caseType.id,
  contact_id: emptyContact.id,
  creator_id: emptyContact.id,
  subject: RECORD_IDENTIFIERS.emptyCaseSubject
});

_.sample(SAMPLE_NUMBER_OF_ACTIVITIES).forEach((index) => createUniqueActivity({
  activity_type_id: 'Follow up',
  case_id: cases[0].id,
  source_contact_id: activeContact.id,
  // We need to avoid issues if we want to add similar activities to other cases:
  subject: `Follow Up ${cases[0].id} - ${index}`,
  activity_date_time: moment().startOf('month').format('YYYY-MM-DD')
}));
```

If we want to make this function less verbose we might need to apply a different kind of strategy and move every single function, not just the cases and activities creation.

Reply:
Thanks. But lets please leave this for future, as I want to merge this PR as soon as possible, and then build on top of it. Its blocking all of our Backstop work at the moment.